lib/phases/cold-launch.js

```diff
@@ -130,36 +133,76 @@ ColdLaunch.prototype.captureEntryMetadata = function(entry) {
  */
 ColdLaunch.prototype.launch = function() {
   var phase = this;
-  var marionette = phase.device.marionette;
-
-  var selector = '#icons > .icon[data-identifier*="' + phase.options.app + '"]';
-
-  if (phase.options.entryPoint) {
-    selector += '[data-identifier*="' + phase.options.entryPoint + '"]';
-  }
 
-  if (phase.options.launchDelay) {
+  if (this.options.launchDelay) {
     debug('Pausing before launching application');
   }
 
+  this.resetInput();
+
   // Delay launch to give time for pre-allocated process and system cool-down
   setTimeout(function() {
-    marionette
-      .startSession()
-      .then(function(client) {
-        marionette.switchToHomescreen();
+    phase.device.input.tap(phase.appX, phase.appY, 1);
+  }, this.options.launchDelay);
+
+  return this.waitForPerformanceEntry('fullyLoaded', this.options.app);
+};
+
+/**
+ * Cache coordinates to the target application
+ * @returns {Promise}
+ */
+ColdLaunch.prototype.getAppCoordinates = function() {
+  var phase = this;
+  var marionette = phase.device.marionette;
+  var selector = '#icons > .icon[data-identifier*="' + this.options.app + '"]';
+
+  if (this.options.entryPoint) {
+    selector += '[data-identifier*="' + this.options.entryPoint + '"]';
+  }
+
+  return marionette
+    .startSession()
+    .then(function(client) {
+      client.switchToFrame();
+
+      // Capture the full height of the System to determine how much shorter
+      // the homescreen is
+      var systemHeight = client.executeScript(function() {
+        return window.wrappedJSObject.innerHeight;
+      });
 
-        client
-          .findElement(selector)
-          .tap(20, 20);
+      marionette.switchToHomescreen();
+
+      var element = client.findElement(selector);
```

Comment:
If you misspell the app name on the command line, this will fail, and raptor just times out with only a timeout error; perhaps there should be a better error message if the app icon cannot be found